public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx

```diff
@@ -131,11 +148,16 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
       return;
     }
 
-    const targetPane = pane === Pane.Top ? 'topPaneSize' : 'rightPaneSize';
     const { updatePanelEditorUIState } = this.props;
-    updatePanelEditorUIState({
-      [targetPane]: size,
-    });
+    if (pane === Pane.Top) {
+      updatePanelEditorUIState({
+        topPaneSize: `${(size / window.innerHeight) * 100}%`,
```

Comment:
I suggest keeping numbers in the state rather than strings. Can't see any benefit in having percentage strings persisted. What is interesting from the component rendering POV is the ratio between the pane size vs window size. This way we can easily switch between % and px if we decide to.